src/state/trajectory/reducer.ts

```diff
@@ -119,13 +134,22 @@ const actionToConfigMap: TypeToDescriptionMap = {
         perform: (
             state: TrajectoryStateBranch,
             action: ReceiveFileToConvertAction
-        ) => ({
-            ...state,
-            processingData: {
-                ...state.processingData,
-                fileToConvert: action.payload,
-            },
-        }),
+        ) => {
+            // prevent side effects if called with no payload
+            // (e.g. when configuring server)
+            if (action.payload === null || action.payload === undefined) {
```

Comment:
I don't understand how this would happen

Reply:
So I went with an approach we we trigger this configure/servercheck logic on page load just to make sure that its all squared away before the user proceeds too far in the conversion process, and so that the modal is raised instantly if they reach that file upload break point. That's why I gave it its own action in the first place.

I took your suggestion of piggybacking that logic on the the receive file action instead, but that means that we could trigger that action without a payload and potentially wipe out some data that was in state by having the payload be null or undefined. It's unlikely given the order of operations but I'm just trying to account for that edge case.

I could just remove the useEffect and not run the server check on page load, or just remove this check and assume this never happens.

Reply:
yeah this isn't needed. We only need to call it once. If you want it to happen at the very beginning of the process you could use the `convert file` action instead, which could be renamed, it's really `initializeFileConversion`. Probably makes the most sense

Reply:
done